src/pkg/services/m365/api/item_pager.go

```diff
@@ -182,29 +188,29 @@ func getAddedAndRemovedItemIDs[T any](
 	prevDeltaLink string,
 	canMakeDeltaQueries bool,
 	aarh addedAndRemovedHandler[T],
-) ([]string, []string, DeltaUpdate, error) {
+) (map[string]time.Time, bool, []string, DeltaUpdate, error) {
```

Comment:
This return is getting a bit choky.  Thoughts on condensing it into something like below on a follow-up?

```go
type addedAndRemoved struct {
  Added map[string]time.Time
  Removed []string
  CanUseModTime bool
  Delta DeltaUpdate
}

func getAddedAndRemovedItetmIDs[T any](...) (addedAndRemoved, error) {...}
```